Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs

```diff
@@ -36,125 +41,348 @@ public class CLIProcessor
-                    GingerGrid gingerGrid = new GingerGrid(15555);   // TODO: get port from CLI arg
-                    gingerGrid.Start();
-
-                    ServiceGridTracker serviceGridTracker = new ServiceGridTracker(gingerGrid);
-
-                    Console.ReadKey();
-                    break;
+                runOptions.Runset = runset;
+            }
+
+            var arguments = Parser.Default.FormatCommandLine<RunOptions>(runOptions);
+
+            string exe = "dotnet GingerConsole.dll ";
 
+            sb.Append(@"Open solution at '" + runOptions.Solution + "'");
+            if (runOptions.Environment == null)
+            {
+                sb.Append(" auto select environment (assuming only one exist)");
+            }
+            else
+            {
+                sb.Append(" select environment '" + runOptions.Environment + "'");
+            }
+                            
+            sb.Append(" and execute run set '").Append(runOptions.Runset).Append("'");
+            sb.Append(Environment.NewLine);
+            sb.Append("Long form : ").Append("'" + exe + arguments + "'").Append(Environment.NewLine);
+            sb.Append("Short form: ").Append("'" + exe + "run");
+            sb.Append(" -").Append(CLIOptionClassHelper.GetAttrShorName<RunOptions>(nameof(RunOptions.Solution))).Append(" ").Append(WrapTextIfNeeded(runOptions.Solution));
+            if (env != null)
+            {
+                sb.Append(" -").Append(CLIOptionClassHelper.GetAttrShorName<RunOptions>(nameof(RunOptions.Environment))).Append(" ").Append(WrapTextIfNeeded(runOptions.Environment));
             }
+
+            if (runset != null)
+            {
+                sb.Append(" -").Append(CLIOptionClassHelper.GetAttrShorName<RunOptions>(nameof(RunOptions.Runset))).Append(" ").Append(WrapTextIfNeeded(runOptions.Runset));
+            }
+
+            sb.Append("'");
+            sb.Append(Environment.NewLine);
+            sb.Append(Environment.NewLine);
+            return sb.ToString();
         }
 
-        private void PerformLoadAndExecution(string configurations)
+        string WrapTextIfNeeded(string val)
         {
-            Reporter.ToLog(eLogLevel.DEBUG, "Loading Configurations...");
-            mCLIHandler.LoadContent(configurations, mCLIHelper, WorkSpace.Instance.RunsetExecutor);
+            if (val.IndexOf(' ') > 0)
+            {
+                return "\"" + val + "\"";
+            }
+            else
+            {
+                return val;
+            }
+        }
+
+        private int HandleSCMOptions(SCMOptions sCmOptions)
+        {
+            SetVerboseLevel(sCmOptions.VerboseLevel);
+            Reporter.ToLog(eLogLevel.INFO, "Running SCM options");
+
+            mCLIHandler = new CLISCM();
+            mCLIHelper.SourceControlURL = sCmOptions.URL;
+            
+            // do the rest !!!
+            
+            ExecuteRunSet();
```

Comment:
HandleSCMOptions() is for source control operations no? so why we do execution in it?

Reply:
you so this comment?